fix: added new constructor for genesis supply that matches upstream

nicolaslara opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
v    If your PR doesn't close an issue, that's OK!  Just remove the Closes: #XXX line!
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

Needed for https://github.com/osmosis-labs/osmosis/issues/2307

## What is the purpose of the change

This is a replacement for https://github.com/osmosis-labs/cosmos-sdk/pull/305 with a cleaner change history. 

The sdk fork is using GenesisSupplyOffsets in the banktypes.NewGenesisState() constructor. This is incompatible with the ibc-go testing package which expects the new signature (https://github.com/cosmos/ibc-go/tree/main/testing).


## Brief Changelog

 - Updated the constructor to match the expected interface.
 - Added a new constructor that can be used for the previous behaviour


## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note
 - Does this pull request introduce a new feature or user-facing behavior changes? (no)
 - Is a relevant changelog entry added to the Unreleased section in CHANGELOG.md? (no)
 - How is the feature or change documented? (not applicable)

nicolaslara committed:
added new constructor for genesis supply that matches upstream

czarcas7ic commented on x/bank/types/genesis.go:
```diff
@@ -63,6 +63,18 @@ func (gs GenesisState) Validate() error {
+	}
+}
+
+// NewGenesisState creates a new genesis state with supply offsets
```
```suggestion
// NewGenesisStateWithSupplyOffsets creates a new genesis state with supply offsets
```

czarcas7ic reviewed (approved):
LGTM

p0mvn commented on x/bank/types/genesis.go:
```diff
@@ -63,6 +63,18 @@ func (gs GenesisState) Validate() error {
+}
+
+// NewGenesisState creates a new genesis state with supply offsets
+func NewGenesisStateWithSupplyOffsets(params Params, balances []Balance, supply sdk.Coins,
```
nit: we can also do a decorator pattern style like here with gov deposit params:
https://github.com/osmosis-labs/cosmos-sdk/blob/c4f55a31756ff8bf63c9c17be75fb5f85c1d580e/x/gov/types/params.go#L94

nicolaslara committed:
moved to decorator pattern

nicolaslara commented:
Changed this to the decorator pattern as requrested by @p0mvn and fixed the bad comment as suggested by @czarcas7ic. Should be ready to merge, but since this change invalidates the review, can I get a 👍 if it's ok to merge this?
